### PR TITLE
New docs and examples

### PR DESCRIPTION
**Pull request by guicassolato**
- [x] new examples
    - simple-api-key
    - alternative-credentials-location
    - forbidden-ip
    - ip-range-allow-list
    - short-lived-api-keys ("beta-testers")
    - read-only-outside
    - kubernetes-auth
    - simple-oidc (w/ keycloak)
    - oidc-active-tokens-only (userinfo)
    - oidc-multiple-sources (keycloak+dex)
    - resource-level-authz ("owned-resources")
    - keycloak-rbac
    - keycloak-abac (the old example from the readme)
    - 8-sample-use-cases (from the slides)
- [x] `examples/README.md`
- [x] remove `dex/README.md` (it is now just another example in `examples/`)
- [x] rename the "Echo API" as "Talker API" (to avoid confusion with so many other Echo APIs)
- [x] `docs/deploy.md` (+ better make targets)
- [x] move CoC to `docs/code_of_conduct.md`
- [x] `docs/contributing.md`
- [x] remove any references to "ostia"
- [x] update main `README.md` accordingly
- [x] PlantUML diagrams

**Best way to review this PR:** start from the new README at https://github.com/3scale-labs/authorino/tree/better-docs-and-examples and follow the links to `/examples` and `/docs/*`.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing to Authorino
+
+## Open issues
+
+Start by checking the list of [issues](https://github.com/3scale-labs/authorino/issues) in GitHub. Sometimes your contribution may have been filed in the past or part of another ongoing conversation.
```
Comment (thomasmaas) [inline docs/contributing.md:5]:
```suggestion
Start by checking the list of [issues](https://github.com/3scale-labs/authorino/issues) in Github. Maybe your idea was discussed in the past or is part of an ongoing conversation.
```

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+If you believe that your contribution is new, please open an issue before sending your pull-request.
```
Comment (thomasmaas) [inline docs/contributing.md:7]: Do we want to say something like" open an issue, describe your idea and start a conversation"?

File: docs/deploy.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+```sh
+git clone git@github.com:3scale-labs/authorino.git && cd authorino
+```
+
+## Meet the requirements
+
+Having a [Kubernetes](https://kubernetes.io/) server up and running is the only actual requirement to deploy Authorino.
+
+With the tools provided in the Authorino repo nevertheless, you can easily have a local cluster setup (using [Kind](https://kind.sigs.k8s.io)). In this case, the only requirement is [Docker](https://docker.com).
```
Comment (thomasmaas) [inline docs/deploy.md:21]: as in, no need for kubernetes?

Comment (thomasmaas) [reply docs/deploy.md:21]: => Nevertheless, with the tools provided in the Authorino repo…

File: docs/deploy.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+### Requirements: the untold truth
+
+Apart from the obvious requirement of having a running instance of a Kubernetes server, Authorino also rely on some other components and capabilities to perform its function. A brief list of such indispensable parts follows:
```
Comment (thomasmaas) [inline docs/deploy.md:25]:
```suggestion
Apart from the obvious requirement of having a running instance of a Kubernetes server, Authorino also relies on some other components and capabilities to perform its function:
```

File: docs/deploy.md (hunk 3 above)
Comment (guicassolato) [reply docs/deploy.md:21]: Correct. Kind starts a Kubernetes cluster in a docker container.

File: docs/contributing.md (hunk 2 above)
Comment (guicassolato) [reply docs/contributing.md]: I hope it's better now.

File: docs/deploy.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+With the tools provided in the Authorino repo nevertheless, you can easily have a local cluster setup (using [Kind](https://kind.sigs.k8s.io)). In this case, the only requirement is [Docker](https://docker.com).
+
+### Requirements: the untold truth
+
+Apart from the obvious requirement of having a running instance of a Kubernetes server, Authorino also relies on some other components and capabilities to perform its function:
+
+<!-- TODO: Add minimum required Kubernetes version -->
+
+- Permission over the Kubernetes server to create Custom Resource Definitions (CRDs). This is actually a one-time requisite, necessary only to install Authorino, and assumed to be known by cluster administrators who should be able to handle as desired through the Kubernetes API Role-Based Access Control bindings.
```
Comment (thomasmaas) [inline docs/deploy.md:29]:
```suggestion
- Permission from the Kubernetes server to create Custom Resource Definitions (CRDs) during Authorino's installation. Cluster administrators can handle this requirement through the Kubernetes API Role-Based Access Control bindings.
```

Comment (thomasmaas) [reply docs/deploy.md:29]: Actually i would consider skipping this whole paragraph as it's true for anything that needs to create CRDs. Or maybe add it to the first part where you talk about k8s.

File: docs/deploy.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+git clone git@github.com:3scale-labs/authorino.git && cd authorino
+```
+
+## Meet the requirements
+
+Having a [Kubernetes](https://kubernetes.io/) server up and running is the only actual requirement to deploy Authorino.
+
+With the tools provided in the Authorino repo nevertheless, you can easily have a local cluster setup (using [Kind](https://kind.sigs.k8s.io)). In this case, the only requirement is [Docker](https://docker.com).
+
+### Requirements: the untold truth
```
Comment (thomasmaas) [inline docs/deploy.md:23]:
```suggestion
### Feature-specific requirements
```

File: docs/deploy.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+Apart from the obvious requirement of having a running instance of a Kubernetes server, Authorino also relies on some other components and capabilities to perform its function:
+
+<!-- TODO: Add minimum required Kubernetes version -->
+
+- Permission over the Kubernetes server to create Custom Resource Definitions (CRDs). This is actually a one-time requisite, necessary only to install Authorino, and assumed to be known by cluster administrators who should be able to handle as desired through the Kubernetes API Role-Based Access Control bindings.
+- [Envoy](https://www.envoyproxy.io) proxy (or, technically, any proxy that implements the client-side of the [gRPC protocol](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/ext_authz). With Envoy, ultimately, virtual hosts will be associated to the upstream APIs to protect, and an [External Authorization Filter](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/ext_authz) should point to Authorino. Check [this example](/examples/envoy/envoy-deploy.yaml) if you need to.
```
Comment (thomasmaas) [inline docs/deploy.md:30]: I would add this paragraph to the first part too and then rename the subtitle as suggested.

File: docs/deploy.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+## Deploy
+
+Choose between the options below to continue:
+
+[**Option A:** Local cluster (try it out and examples)](#option-a-local-cluster)<br/>
+[**Option B:** Custom deployment](#option-b-custom-deployment)
+
+### Option A: Local cluster
+
+Option A is essentially meant for trying out Authorino locally. It brings a bundle of a [Kind](https://kind.sigs.k8s.io)-managed Kubernetes cluster, with a freshly built Authorino image loaded up and pre-configured sample resources to make testing and trying out simple and easy.
```
Comment (thomasmaas) [inline docs/deploy.md:46]:
```suggestion
Option A is meant for trying out Authorino locally. It gives you a bundle of a [Kind](https://kind.sigs.k8s.io)-managed Kubernetes cluster, with a freshly built Authorino image and pre-configured sample resources. 
```

**Review by thomasmaas — CHANGES_REQUESTED** [review]

File: docs/deploy.md (hunk 5 above)
Comment (guicassolato) [reply docs/deploy.md]: I will keep it (moving up as you suggested). I think reminding about having privileges to create CRDs as a requirement won't harm.

**Review by thomasmaas — APPROVED** [review]
Let's keep improving but also keep the speed.

**Review by thomasmaas — APPROVED** [review]

**Comment by cntlsn** [comment]
@guicassolato the link to the README in the description of this PR is broken (https://github.com/3scale-labs/authorino/tree/better-docs-and-examples). How can we get access to it?

**Pull request merged**